rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -215,7 +215,14 @@ public void filter(ContainerRequestContext containerRequestContext) throws IOExc
         }
 
         enlist = true;
-        nested = resourceInfo.getResourceMethod().isAnnotationPresent(NestedLRA.class);
+        nested = type == LRA.Type.NESTED;
+        if (nested) {
+            type = LRA.Type.REQUIRED;
+
+            if (incommingLRA != null) {
+                headers.putSingle(LRA_HTTP_PARENT_CONTEXT_HEADER, incommingLRA.toASCIIString());
+            }
+        }
 
         switch (type) {
             case MANDATORY: // a txn must be present
```

Comment:
if type is `LRA.Type.NESTED` so `nested == true` none of these switch cases will ever be entered so I don't see any reason for keeping `if (nested)` blocks inside these cases (e.g. line 224). I would also prefer to move `if (nested)` block from line 218 to a new case branch in this switch so the approach is unified.

Reply:
We may need to revisit the ability to set attributes on nested LRAs so I don't see any harm in leaving the tests. When/if we need to look at performance we can look into it then.

Reply:
sorry, I don't understand. We've changed the spec so now nested LRA is defined by type and there will never be a case that the type will be MANDATORY and NESTED at the same time so this nested ifs are never to be executed.

Reply:
@mmusgrov: I agree with @xstefank . I haven't recognized this relation to the spec. Re-factoring the switch would make the code nicer and enhance the readability. I really consider such clean-up as a  benefit. (In case there is a some change of the spec in future there is still the history in the git.)

Reply:
My concern is readability and understandability of the code. Layering the clean-up as a next commit sounds good to me.